### PR TITLE
Rank recipients: remove unnecessary feedback paths #8087

### LINKED ISSUE #8087: Rank recipients: remove unnecessary feedback paths
For Rank (recipients) question, it doesn't make sense to have feedback paths where the recipient is `Self` or `None`. Remove/disable these feedback paths for rank recipients.

### PR DESCRIPTION
**Pull request by VamsiSangam**
Fixes #8087

### CODE REVIEW

**Comment by VamsiSangam** [comment]
New UI -

5cnbvfjt4g

Ready for review.

File: src/test/resources/pages/instructorStudentRecordsPageEditedCommentOfDifferentInstructor.html [hunk 1]
```diff
@@ -402,7 +402,7 @@ <h4 class="modal-title">
                                   </table>
                                 </div>
                                 <div class="form-group">
-                                  <div class="panel panel-default panel-body mce-content-body content-editor" contenteditable="true" id="responsecommenttext-RGQ-1-1-1-2" spellcheck="false" style="position: relative;">
+                                  <div class="panel panel-default panel-body mce-content-body content-editor empty" contenteditable="true" id="responsecommenttext-RGQ-1-1-1-2" spellcheck="false" style="position: relative;">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorStudentRecordsPageEditedCommentOfDifferentInstructor.html]: Again, this is an unrelated change. Revert it.

File: src/main/webapp/dev/js/common/questionRank.es6 [hunk 2]
```diff
@@ -220,10 +220,18 @@ function bindRankEvents() {
         });
 }
 
+function removeInvalidFeedbackPathsForRankRecipientQuestions() {
```
Comment (wkurniawan07) [inline src/main/webapp/dev/js/common/questionRank.es6]: Maybe I miss something, but how does this code know to target only "rank recipients" type of question and not including "rank options"?

Comment (VamsiSangam) [reply src/main/webapp/dev/js/common/questionRank.es6]: Sorry about that 😅 Will fix it!

**Comment by VamsiSangam** [comment]
@wkurniawan07 I believe the build errors are due to an unrelated issue. Can you help me with it?

File: src/main/webapp/dev/js/common/questionRank.es6 [hunk 3]
```diff
@@ -220,10 +220,21 @@ function bindRankEvents() {
...
+    $('input[name="questiontype"][value="RANK_RECIPIENTS"]').each(function () {
+        const $form = $(this).closest('form');
+
+        $form.find('[data-recipient-type="NONE"],[data-recipient-type="SELF"]').hide();
```
Comment (wkurniawan07) [inline src/main/webapp/dev/js/common/questionRank.es6]: There's no `OWN_TEAM` here?

Comment (VamsiSangam) [reply src/main/webapp/dev/js/common/questionRank.es6]: Yes there should be `OWN_TEAM` here. Will correct it.

Comment (VamsiSangam) [reply src/main/webapp/dev/js/common/questionRank.es6]: Upon closer look I saw that there's no need for `[data-recipient-type="OWN_TEAM"]` here because that statement was to exclude invalid recipients selected through the common feedback paths button, it doesn't have a path where recipient is `own team`. So `OWN_TEAM` isn't needed here.

**Comment by VamsiSangam** [comment]
> Does this work on newly added rank recipient questions as well?

I couldn't quite understand this. This PR is a UI check. So, new rank recipient questions added after this PR is merged will have limited feedback path options, but the existing rank recipients questions will remain unaffected.

Given that, I think I must check if old feedback path option is displayed for existing questions (as we are hiding them in this PR) -> will update you on this.

**Comment by wkurniawan07** [comment]
@VamsiSangam, I was not referring to questions already existing. I was referring to new questions added with the "Add Question" button. Existing questions being loaded in the edit page should work, as evident from the UI check.

**Comment by VamsiSangam** [comment]
That is a NO 😅 . I think [this](https://github.com/VamsiSangam/teammates/blob/8087-rank-recipients-remove-feedback-paths/src/test/resources/pages/instructorFeedbackRankMinMaxChoicesSuccess.html#L2628) and the way I select rank recipient question proves it.

There will be little complication because all new question frames share the feedback path button. So I'll have to hide invalid paths when rank recipients is selected and again display them if anything else is selected.

Will work on it.

**Comment by VamsiSangam** [comment]
Ready for review.

File: src/test/resources/pages/instructorFeedbackQuestionVisibilityOptions.html [hunk 4]
```diff
@@ -1011,7 +1011,7 @@ <h5 class="col-sm-2">
                   <option selected="" value="TEAMS">
                     Other teams in the course
                   </option>
-                  <option style="display: none;" value="OWN_TEAM">
+                  <option style="display: block;" value="OWN_TEAM">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackQuestionVisibilityOptions.html]: Looks like this is unintended? This question has feedback path from "Teams in this course > ..." which should mean "Giver's team" cannot be selected.

File: src/test/resources/pages/instructorFeedbackQuestionVisibilityPreview.html [hunk 5]
```diff
@@ -1011,7 +1011,7 @@ <h5 class="col-sm-2">
                   <option selected="" value="TEAMS">
                     Other teams in the course
                   </option>
-                  <option style="display: none;" value="OWN_TEAM">
+                  <option style="display: block;" value="OWN_TEAM">
```
Comment (wkurniawan07) [inline src/test/resources/pages/instructorFeedbackQuestionVisibilityPreview.html]: Similarly here.

**Comment by wkurniawan07** [comment]
Hi @VamsiSangam, any updates here?

**Comment by wkurniawan07** [comment]
Closing due to inactivity for a long period. @VamsiSangam feel free to reopen when you're ready to resume the work here.
